File: 1688.count-of-matches-in-tournament.py
#
# @lc app=leetcode id=1688 lang=python
#
# [1688] Count of Matches in Tournament
#

# @lc code=start
class Solution(object):
    def numberOfMatches(self, n):
        """
        :type n: int
        :rtype: int
        """
        matches = 0
        while n != 1:
            if n%2 == 1:
                matches += (n-1)//2
                n = (n - 1)//2 + 1
            else:
                matches += n//2
                n = n//2
        return matches
        
# @lc code=end

File: 1688.count-of-matches-in-tournament.java
/*
 * @lc app=leetcode id=1688 lang=java
 *
 * [1688] Count of Matches in Tournament
 */

// @lc code=start
class Solution {
    public int numberOfMatches(int n) {
        if(n == 1){
            return 0;
        }
        if(n % 2 == 0){
            return (n/2) + numberOfMatches(n/2);
        }
        else{

            return ((n-1)/2) + numberOfMatches(((n-1)/2) + 1);
        }
    }
}
// @lc code=end

